lib/ansible/plugins/inventory/vmware_vm_inventory.py

```diff
@@ -302,22 +304,6 @@ class InventoryModule(BaseInventoryPlugin, Cacheable):
 
     NAME = 'vmware_vm_inventory'
 
-    def verify_file(self, path):
```

Comment:
Isn't this method required to work ?

Reply:
Not required as it is implemented in the `auto` plugin used to route to the `vmware_vm_inventory` plugin so the redefining it here is redundant.

Reply:
All plugins should implement their own `verify_file` so that the auto plugin knows whether or not a file is valid for the individual plugin.

Reply:
@sivel what defines "valid"? As the method was implemented, it just checked to see if the file was `vmware.yml|yaml`. I believe that is the check that the auto plugin is already doing.

also, with the plugin `NAME=vmware_vm_inventory` and the verify_file method checking for `vmware.yml|yaml` I think this causes a conflict where either auto would fail or the plugin verify_file would fail.

Reply:
That is largely the goal of that function. The file must still be named in a way that ends with `vmware.yaml` or `vmware.yml`.  All of the plugins implement this mechanism and check for a namespaced filename.

The `auto` plugin checks that they end with `yaml` or `yml`, the individual plugins implement more specific checks.

I didn't write the behavior, so I cannot speak really more to it.  @s-hertel might be able to provide more insight.